pkg/cmd/target/kubernetes/create/create.go

```diff
@@ -0,0 +1,1144 @@
+		Example: heredoc.Docf("$ %s deployment-target kubernetes create", constants.ExecutableName),
+		Aliases: []string{"new"},
+		RunE: func(c *cobra.Command, _ []string) error {
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this deployment target.")
+	flags.StringVar(&createFlags.AuthenticationType.Value, createFlags.AuthenticationType.Name, "", "The authentication type to use.")
+	flags.StringVar(&createFlags.Account.Value, createFlags.Account.Name, "", "The name of the account to use for authentication.")
+
+	// Azure Service Principal
+	flags.StringVar(&createFlags.AKSClusterName.Value, createFlags.AKSClusterName.Name, "", "The AKS cluster name.")
+	flags.StringVar(&createFlags.AKSResourceGroupName.Value, createFlags.AKSResourceGroupName.Name, "", "The AKS resource group name.")
+	flags.BoolVar(&createFlags.UseAdminCredentials.Value, createFlags.UseAdminCredentials.Name, false, "Enabling this option passes the --admin flag to az aks get-credentials. This is useful for AKS clusters with Azure Active Directory integration.")
+
+	// AWS Account
+	flags.BoolVar(&createFlags.UseServiceRole.Value, createFlags.UseServiceRole.Name, false, "Execute using the AWS service role for an EC2 instance.")
+	flags.BoolVar(&createFlags.AssumeServiceRole.Value, createFlags.AssumeServiceRole.Name, false, "Assume a different AWS service role.")
+	flags.StringVar(&createFlags.AssumedRoleARN.Value, createFlags.AssumedRoleARN.Name, "", "ARN of assumed AWS service role.")
+	flags.StringVar(&createFlags.AssumedRoleSessionName.Value, createFlags.AssumedRoleSessionName.Name, "", "Session name of assumed AWS service role.")
+	flags.IntVar(&createFlags.AssumedRoleSessionDuration.Value, createFlags.AssumedRoleSessionDuration.Name, 3600, "AWS assumed role session duration in seconds. (defaults to 3600 seconds, 1 hour)")
+	flags.StringVar(&createFlags.AssumedRoleExternalID.Value, createFlags.AssumedRoleExternalID.Name, "", "AWS assumed role external ID.")
+	flags.StringVar(&createFlags.EKSClusterName.Value, createFlags.EKSClusterName.Name, "", "AWS EKS Cluster Name")
+
+	// Google Cloud Account
+	flags.BoolVar(&createFlags.UseVMServiceAccount.Value, createFlags.UseVMServiceAccount.Name, false, "When running in a Compute Engine virtual machine, use the associated VM service account.")
+	flags.BoolVar(&createFlags.ImpersonateServiceAccount.Value, createFlags.ImpersonateServiceAccount.Name, false, "Impersonate service accounts.")
+	flags.StringVar(&createFlags.ServiceAccountEmails.Value, createFlags.ServiceAccountEmails.Name, "", "Service Account Email.")
+	flags.StringVar(&createFlags.GKEClusterName.Value, createFlags.GKEClusterName.Name, "", "GKE Cluster Name.")
+	flags.StringVar(&createFlags.Project.Value, createFlags.Project.Name, "", "Project.")
```

Comment:
Can we make this a bit more descriptive

Reply:
The flag names will now help give this more meaning. The portal does not provide any more information about them, they are concepts for the targeted platform, i'm not sure how to make it more descriptive.